### PR TITLE
Support HAProxy's PROXY protocol v1 in DialProxy.

### PR DESCRIPTION
**Pull request by danderson**

### CODE REVIEW

**Comment by danderson** [comment]
Travis build horked because of lint (see other PR), but new test passes.

File: tcpproxy.go [hunk 1]
```diff
@@ -284,6 +285,15 @@ type DialProxy struct {
 	// If nil, the error is logged and src is closed.
 	// If non-nil, src is not closed automatically.
 	OnDialError func(src net.Conn, dstDialErr error)
+
+	// ProxyProtocolVersion optionally specifies the version of
+	// HAProxy's PROXY protocol to use. The PROXY protocol provides
+	// connection metadata to the DialProxy target, via a header
+	// inserted ahead of the client's traffic. The DialProxy target
+	// must explicitly support and expect the PROXY header, there is
```
Comment (bradfitz) [inline tcpproxy.go]: semicolon instead of comma

**Review by bradfitz — APPROVED** [review]
LGTM

File: tcpproxy.go (hunk 1 above)
Comment (danderson) [reply tcpproxy.go]: Done.

**Pull request merged**